server/mediaserver.go

```diff
@@ -41,36 +42,39 @@ var ErrRTMPPlay = errors.New("ErrRTMPPlay")
 var ErrRoundInit = errors.New("ErrRoundInit")
 var ErrStorage = errors.New("ErrStorage")
 var ErrDiscovery = errors.New("ErrDiscovery")
+var ErrUnknownStream = errors.New("ErrUnknownStream")
 
 const HLSWaitInterval = time.Second
 const HLSBufferCap = uint(43200) //12 hrs assuming 1s segment
 const HLSBufferWindow = uint(5)
+const StreamKeyBytes = 6
```

Comment:
No reason. Should it be 4? 5? 8? 32?

The stream key only makes it harder for an attacker to **read** the incoming RTMP. A typical node really should have monitoring for repeated attempts in case someone tries to brute force things, eg fail2ban. However, at the moment, we don't really have the logging granularity to facilitate this monitoring, nor do we expose enough information to the goclient to determine data like incoming RTMP IP addresses. We should though.

Reply:
Got it - seems fine as long as we have a constant for it!

Reply:
We'd still want to use our RNG helpers to generate the stream key. It seems that something else also mutates the RNG state during testing, so resetting the seed alone is not the best way to ensure deterministic results during testing. 

Also see https://github.com/livepeer/go-livepeer/pull/625/commits/ccad74cd1aae1d472faa491b697cab9b001ebb3d

Reply:
Wouldn't the built-in helpers from the `rand` package (i.e. `rand.Uint64()`) be relying on the same random seed (since they rely on the same default source) as `RandomIdGenerator` (https://github.com/livepeer/go-livepeer/blob/master/core/streamdata.go#L49) since this function is using `rand.Uint32()` under the hood?

Reply:
Correct, and we can't rely on resetting the seed to give us consistent behavior during unit tests. Hence we want to generate the stream key via the monkey-patchable `RandomIdGenerator` that's introduced above, to get deterministic values for testing.

Reply:
Ah I see - I initially missed your point about monkey patching `RandomIdGenerator`. 👍